File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.js
/**
 * @param {number[]} cardPoints
 * @param {number} k
 * @return {number}
 */
var maxScore = function(cardPoints, k) {


    let n = cardPoints.length
    let sum = 0;
    let max = 0;


    for(let i = n-k ; i<n ; i++){
        sum+=cardPoints[i]
    }
     
     max = sum;

    for(let i=0 ; i<k ; i++){
        sum = sum + cardPoints[i] - cardPoints[n-k + i];
        max = Math.max(sum , max)

    }

    return max
    
};

File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.py
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:

       
        

        current_sum = sum(cardPoints[-k:])

        max_sum = current_sum
       
        for i in range(k):
            current_sum = current_sum + cardPoints[i] - cardPoints[len(cardPoints)-k + i]
            max_sum = max(current_sum , max_sum)


        return max_sum






        